autokeras/hypermodels/preprocessing.py

```diff
@@ -0,0 +1,202 @@
+            as a range for picking a contrast shift value from. Default to 0.5.
+        translation: Boolean. Whether to translate the image.
+        horizontal_flip: Boolean. Whether to flip the image horizontally.
+        vertical_flip: Boolean. Whether to flip the image vertically.
+        gaussian_noise: Boolean. Whether to add gaussian noise to the image.
+    """
+
+    def __init__(self,
+                 percentage=0.25,
+                 rotation_range=180,
+                 random_crop=True,
+                 brightness_range=0.5,
+                 saturation_range=0.5,
+                 contrast_range=0.5,
+                 translation=True,
+                 horizontal_flip=True,
+                 vertical_flip=True,
+                 gaussian_noise=True,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.percentage = percentage
+        self.rotation_range = rotation_range
+        self._rotate_choices = [0]
+        if self.rotation_range == 90:
+            self._rotate_choices = [0, 1, 3]
+        elif self.rotation_range == 180:
+            self._rotate_choices = [0, 1, 2, 3]
+        self.random_crop = random_crop
+        if self.random_crop:
+            # Generate 20 crop settings, ranging from a 1% to 20% crop.
+            self.scales = list(np.arange(0.8, 1.0, 0.01))
+            self.boxes = np.zeros((len(self.scales), 4))
+            for i, scale in enumerate(self.scales):
+                x1 = y1 = 0.5 - (0.5 * scale)
+                x2 = y2 = 0.5 + (0.5 * scale)
+                self.boxes[i] = [x1, y1, x2, y2]
+        self.brightness_range = brightness_range
+        self.saturation_range = self._get_min_and_max(saturation_range,
+                                                      'saturation_range')
+        self.contrast_range = self._get_min_and_max(contrast_range,
+                                                    'contrast_range')
+        self.translation = translation
+        self.horizontal_flip = horizontal_flip
+        self.vertical_flip = vertical_flip
+        self.gaussian_noise = gaussian_noise
+        self.shape = None
+
+    @staticmethod
+    def _get_min_and_max(value, name):
+        if isinstance(value, (tuple, list)) and len(value) == 2:
+            min_value, max_value = value
+            return min_value, max_value
+        elif isinstance(value, (int, float)):
+            min_value = 1. - value
+            max_value = 1. + value
+            return min_value, max_value
+        elif value == 0:
+            return None
+        else:
+            raise ValueError('Expected {name} to be either a float between 0 and 1, '
+                             'or a tuple of 2 floats between 0 and 1, '
+                             'but got {value}'.format(name=name, value=value))
+
+    def build(self, hp, inputs=None):
+        return inputs
+
+
+class FeatureEncoding(block_module.Block):
```

Comment:
"FeatureEncoding" is very generic (everything in ML is a feature and needs encoding, since it usually doesn't start out as a numerical array). Perhaps CategoricalEncoding or CategoricalFeatureEncoding would be more specific? Or CategoricalToNumerical?

Reply:
Right, `CategoricalToNumerical` would be more accurate and more consistent with other preprocessing blocks like `TextToNgramVector`.